de.metas.contracts/src/main/java/de/metas/contracts/interceptor/C_Order.java

```diff
@@ -48,83 +48,40 @@
 import de.metas.invoicecandidate.model.I_C_Invoice_Candidate;
 import de.metas.logging.LogManager;
 
-public class C_Order implements ModelValidator
+@Interceptor(I_C_Order.class)
+public class C_Order
 {
 	private static final Logger logger = LogManager.getLogger(C_OrderLine.class);
 
 	private static final String MSG_ORDER_DATE_ORDERED_CHANGE_FORBIDDEN_1P = "Order_DateOrdered_Change_Forbidden";
 
-	private int m_AD_Client_ID = -1;
-
-	@Override
-	public int getAD_Client_ID()
-	{
-		return m_AD_Client_ID;
-	}
-
-	@Override
-	public void initialize(final ModelValidationEngine engine, final MClient client)
-	{
-		if (client != null)
-			m_AD_Client_ID = client.getAD_Client_ID();
-
-		engine.addModelChange(I_C_Order.Table_Name, this);
-		engine.addDocValidate(I_C_Order.Table_Name, this);
-	}
-
-	@Override
-	public String login(final int AD_Org_ID, final int AD_Role_ID, final int AD_User_ID)
-	{
-		return null;
-	}
-
-	@Override
-	public String modelChange(final PO po, final int type) throws Exception
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE })
+	public void updateDataEntry(final I_C_Order order)
 	{
-		if (type != TYPE_BEFORE_CHANGE)
+		if (InterfaceWrapperHelper.isValueChanged(order, I_C_Order.COLUMNNAME_DateOrdered))
```

Comment:
i think you can use ignoreColumnsChanged within the ModelChange annotation for this